ganalytics.php

```diff
@@ -122,20 +122,94 @@ public function onPluginsInitialized()
+            return ($this->inIPAdressRange($canonicalAddress, "10.0.0.0-10.255.255.255")
+                || $this->inIPAdressRange($canonicalAddress, "172.16.0.0-172.31.255.255")
+                || $this->inIPAdressRange($canonicalAddress, "192.168.0.0-192.168.255.255"));
+        } elseif ($range === 'loopback') {  // RFC 6890
+            return ($this->inIPAdressRange($canonicalAddress, "127.0.0.0-127.255.255.255"));
+        } elseif ($range === 'link-local') {  // RFC 6890
+            return ($this->inIPAdressRange($canonicalAddress, "169.254.0.0-169.254.255.255"));
+        } else {
+            $rangeLimits = explode('-', $range);
+            if (count($rangeLimits) == 2
+                && strcmp($canonicalAddress, $this->canonicalIPAddress($rangeLimits[0])) >= 0
+                && strcmp($canonicalAddress, $this->canonicalIPAddress($rangeLimits[1])) <= 0) {
+                return TRUE;
+            }
+        }
+
+        return FALSE;
+    }
+
+    /**
+     * Documents the reason for blocking GA tracking in a JavaScript comment
+     * @param string $reason
+     */
+    private function documentBlockingReason($reason)
+    {
+        $this->grav['assets']->addInlineJs("/* GA tracking blocked, reason: $reason */");
+    }
+
+    /**
      * Add GA tracking JS when the assets are initialized
      */
     public function onAssetsInitialized()
     {
         // Don't proceed if we are in the admin plugin
-        if ($this->isAdmin()) return;
+        if ($this->isAdmin()) {
+            $this->documentBlockingReason('admin plugin active');
+            return;
+        }
 
         // Don't proceed if there is no GA Tracking ID
         $trackingId = trim($this->config->get('plugins.ganalytics.trackingId', ''));
-        if (empty($trackingId)) return;
+        if (empty($trackingId)) {
+            $this->documentBlockingReason('trackingId not configured');
+            return;
+        }
+
+        // Don't proceed if a blocking cookie is set
+        $blockingCookieName = $this->config->get('plugins.ganalytics.blockingCookie', '');
+        if (!empty($blockingCookieName) && !empty($_COOKIE[$blockingCookieName])) {
+            $this->documentBlockingReason("blocking cookie \"$blockingCookieName\" is set");
+            return;
+        }
 
         // Don't proceed if the IP address is blocked
         $blockedIps = $this->config->get('plugins.ganalytics.blockedIps', []);
-        if (in_array($_SERVER['REMOTE_ADDR'], $blockedIps)) return;
+        if (in_array($_SERVER['REMOTE_ADDR'], $blockedIps)) {
+            $this->documentBlockingReason("client ip " . $_SERVER['REMOTE_ADDR'] . " is in blockedIps");
+            return;
+        }
+
+        // Don't proceed if the IP address is within a blocked range
+        $canonicalClientIpAddress = $this->canonicalIPAddress($_SERVER['REMOTE_ADDR']);
+        $blockedIpRanges = $this->config->get('plugins.ganalytics.blockedIpRanges', []);
```

Comment:
You forgot to implement this paramter in the [blueprint.yaml](https://github.com/escopecz/grav-ganalytics/blob/master/blueprints.yaml).
Additionally you should be able to enable/disable this feature.